### PR TITLE
Add socks4 support 

### PR DESCRIPTION
**Pull request by Airkek**
image

### CODE REVIEW

File: src/proxy.rs [hunk 1]
```diff
@@ -192,8 +198,8 @@ mod tests {
 
     #[cfg(feature = "socks-proxy")]
     #[test]
-    fn parse_proxy_socks_user_pass_server_port() {
-        let proxy = Proxy::new("socks://user:p@ssw0rd@localhost:9999").unwrap();
+    fn parse_proxy_socks4_user_pass_server_port() {
+        let proxy = Proxy::new("socks4://user:p@ssw0rd@localhost:9999").unwrap();
```
Comment (algesten) [inline src/proxy.rs:202]: I think we need to duplicate this test to have a `socks://` ensure SOCKS5 and a `socks4://` doing SOCKS4.

To maintain backwards compatibility and not have to major version, we need to make it so `socks://` means SOCKS5. This should also be highlighted in the docs.

**Review by algesten — CHANGES_REQUESTED** [review]
Hi! Welcome to ureq. This overall looks good. Thanks!

I'm not very good with proxies, cause I never use them. I think there's one little thing we need to address with the PR. See comment.

File: src/proxy.rs (hunk 1 above)
Comment (Airkek) [reply src/proxy.rs:202]: > To maintain backwards compatibility and not have to major version, we need to make it so socks:// means SOCKS5.

`socks` still means SOCKS5

Comment (algesten) [reply src/proxy.rs:202]: Cool! thanks!

**Review by algesten — APPROVED** [review]

**Comment by algesten** [comment]
Perfect! Let's land this. Thanks!

**Pull request merged**